dialogue-core/src/main/java/com/palantir/dialogue/core/DialogueTracedRequestChannel.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.dialogue.core;
+
+import com.google.common.util.concurrent.ListenableFuture;
+import com.palantir.dialogue.Channel;
+import com.palantir.dialogue.Endpoint;
+import com.palantir.dialogue.Request;
+import com.palantir.dialogue.Response;
+import com.palantir.tracing.Tracers;
+
+public final class DialogueTracedRequestChannel implements Channel {
+    private final Channel delegate;
+
+    public DialogueTracedRequestChannel(Channel delegate) {
+        this.delegate = delegate;
+    }
+
+    @Override
+    public ListenableFuture<Response> execute(Endpoint endpoint, Request request) {
+        String operationName = "Dialogue: request " + endpoint.serviceName() + "#" + endpoint.endpointName();
+        return Tracers.wrapListenableFuture(operationName, () -> delegate.execute(endpoint, request));
```

Comment:
@ferozco I think previously we included the HTTP verb too..

Reply:
https://github.com/palantir/dialogue/pull/582#discussion_r399687592

Reply:
fair enough